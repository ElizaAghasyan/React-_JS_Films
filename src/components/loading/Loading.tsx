import { useState } from 'react';
import styles from './Loading.module.scss'

const LoadingDots = () => {
    const [loading, _] = useState(true);

    return (
        <div className={styles.dots}>
            {loading ?(
                <>
                    <div className={styles.spinner}>
                        <div className={styles.dot1}> </div>
                        <div className={styles.dot2}> </div>
                        <div className={styles.dot3}> </div>
                    </div>
                    <p>LOADING</p>
                </>
            ) : null}
        </div>
    );
}

export default LoadingDots;
