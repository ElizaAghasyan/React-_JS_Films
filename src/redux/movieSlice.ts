import {createAsyncThunk, createSlice} from "@reduxjs/toolkit";
import serverApi, {movieType} from "../config/movieApi";

const initialState = {
    movies: {},
}

interface MovieType  {
    id: number;
    title: string;
    overview: string;
    poster_path: string;
    vote_average: number;
    backdrop_path: string;
    vote_count: number;
    name: string;
}


export const fetchAsyncMovies = createAsyncThunk('movies/fetchAsyncMovies', async () => {
    const params = {page: 1}
    let response = await serverApi.getMovieList(movieType.popular, {params})
        .catch(err => {
            console.log(err)
        })
    let result = await response?.data.results[Math.floor(Math.random() * response.data.results.length - 1)]
    return result;
});


export const movieSlice = createSlice({
    name: 'movies',
    initialState,
    reducers: {
        addMovies: (state, {payload}) => {
            state.movies = payload
        }
    },
    extraReducers: (builder) => {
        builder.addCase(fetchAsyncMovies.pending, (state, action) => {
            console.log("Pending...")
        })
        builder.addCase(fetchAsyncMovies.fulfilled, (state: {}, action) => {
            return {
                ...state, movies: action.payload
            }
        })
        builder.addCase(fetchAsyncMovies.rejected, (state, action) => {
            console.log("Pending...")
        })
    }
});

export const {addMovies} = movieSlice.actions;

export const getBannerMovie = (state: { movies: { movies: MovieType }}) => state.movies.movies;


export default movieSlice.reducer;
