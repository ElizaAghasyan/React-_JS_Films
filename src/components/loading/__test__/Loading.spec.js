import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "../../../redux/store";
import Loading from '../Loading';

describe('Loading Component', () => {

    it('Loading snapshot test', () => {
        const component = renderer.create(<Provider store={store}><Loading /></Provider>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
