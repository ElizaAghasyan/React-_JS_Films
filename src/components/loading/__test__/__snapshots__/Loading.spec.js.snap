// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component Loading snapshot test 1`] = `
<div
  className="dots"
>
  <div
    className="spinner"
  >
    <div
      className="bounce1"
    >
       
    </div>
    <div
      className="bounce2"
    >
       
    </div>
    <div
      className="bounce3"
    >
       
    </div>
  </div>
  <p>
    LOADING
  </p>
</div>
`;
